### ISSUE TITLE
Failing build due to using nativescript-sqlite with latest instead with numeric version

### ISSUE THREAD

**Issue by NickIliev** [Kinvey/nativescript-sdk#48]
The Kinvey plugin is using [nativescript-sqlite as a dependency](https://github.com/Kinvey/nativescript-sdk/blob/master/package.json#L12). However, the dependency version is set to `latest` which will cause a duplicate error and failed build when we have an application that is also using nativescript-sqlite (but with numeric version).

e.g. use this app and try to build for iOS (`tns run ios`). This wil result the following error
```
Xcode build...
warning: duplicate output file '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/F
rameworks/sqlitewrapper.framework' on task: PBXCp /Users/niliev/Downloads/issue-359/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework /Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework (in target 'issue359')
warning: duplicate output file '' on task: PBXCp /Users/niliev/Downloads/issue-359/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework /Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework (in target 'issue359')
error: Multiple commands produce '<PbxCp /Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework>':
1) Target 'issue359' (project 'issue359') has copy command from '/Users/niliev/Downloads/issue-359/node_modules/kinvey-nativescript-sdk/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework' to '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework'
2) Target 'issue359' (project 'issue359') has copy command from '/Users/niliev/Downloads/issue-359/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework' to '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework'
error: Multiple commands produce '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework':
1) Target 'issue359' (project 'issue359') has copy command from '/Users/niliev/Downloads/issue-359/node_modules/kinvey-nativescript-sdk/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework' to '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/InstallationBuildProductsLocation/Applications/issue359.app/Frameworks/sqlitewrapper.framework'
2) Target 'issue359' (project 'issue359') has copy command from '/Users/niliev/Downloads/issue-359/node_modules/nativescript-sqlite/platforms/ios/sqlitewrapper.framework' to '/Users/niliev/Library/Developer/Xcode/DerivedData/issue359-hgthoerwdxbmgrduguzcyapyignk/Build/Intermediates.noindex/ArchiveIntermediates/issue359/Installa
```

The solution would be to [replace `latest`](https://github.com/Kinvey/nativescript-sdk/blob/master/package.json#L12) with numeric versioning(e.g. by using ^ which will need update only after major versions)

e.g.
```
"nativescript-sqlite": "^2.3.3",

```

**Comment by NickIliev** [comment]
On second though perhaps the `nativescript-sqlite` should be a **peerDependency**

**Comment by bilger-progress** [comment]
The issue has been reported. Will keep this ticket up-to-date. Thanks!

**Comment by bilger-progress** [comment]
Issue fixed and deployed with version 4.0.0
https://github.com/Kinvey/js-sdk/tree/master/packages/nativescript-sdk
